benchmark/sqllogictest_tester/run_tester.sh

```diff
@@ -0,0 +1,273 @@
+#!/bin/bash
+
+set -eo pipefail
+
+function fail() {
+    1>&2 echo "$@"
+    exit 1
+}
+
+base_dir=$(cd ../../ && pwd)
+dsp_dir=$(pwd)
+log_dir="$dsp_dir"/tempLogs
+sqllogictest_checkout="$base_dir/sqllogictest"
+logictest="$base_dir/tempDolt/go/libraries/doltcore/sqle/logictest"
+logictest_main="$logictest"/main
+schema_dir="$base_dir/go/libraries/doltcore/sqle/logictest"
+old_path=$PATH
+
+if [[ "$#" -ne 1 ]]; then
+    fail Usage: ./run_regressions.sh ENV_VARIABLES_FILE
+fi
+
+source "$1"
+if [ -z "$DOLT_ROOT_PATH" ]; then fail Must supply DOLT_ROOT_PATH; fi
+if [ -z "$DOLT_CONFIG_PATH" ]; then fail Must supply DOLT_CONFIG_PATH; fi
+if [ -z "$DOLT_GLOBAL_CONFIG" ]; then fail Must supply DOLT_GLOBAL_CONFIG; fi
+if [ -z "$CREDSDIR" ]; then fail Must supply CREDSDIR; fi
+if [ -z "$DOLT_CREDS" ]; then fail Must supply DOLT_CREDS; fi
+if [ -z "$CREDS_HASH" ]; then fail Must supply CREDS_HASH; fi
+if [ -z "$TMP_TESTING_DIR" ]; then fail Must supply TMP_TESTING_DIR; fi
+if [ -z "$TMP_CSV_DIR" ]; then fail Must supply TMP_CSV_DIR; fi
+if [ -z "$TEST_FILE_DIR_LIST" ]; then fail Must supply TEST_FILE_DIR_LIST; fi
+if [ -z "$COMMITS_TO_TEST" ]; then fail Must supply COMMITS_TO_TEST; fi
+if [ -z "$DOLT_BRANCH" ]; then fail Must supply DOLT_BRANCH; fi
+if [ -z "$TEST_N_TIMES" ]; then fail Must supply TEST_N_TIMES; fi
+
+[[ "$TEST_N_TIMES" =~ ^[0-9]+$ ]] || fail TEST_N_TIMES must be a number
+
+function setup() {
+    rm -rf "$CREDSDIR"
+    mkdir -p "$CREDSDIR"
+    dolt config --global --add metrics.disabled true
+    dolt creds import "$DOLT_CREDS"
+    dolt version
+}
+
+function setup_testing_dir() {
+    rm -rf "$TMP_TESTING_DIR"
+    mkdir -p "$TMP_TESTING_DIR"
+
+    IFS=', ' read -r -a test_list <<< "$TEST_FILE_DIR_LIST"
+    for fd in "${test_list[@]}"
+    do
+         dir=$(dirname "$fd")
+         mkdir -p "$TMP_TESTING_DIR"/"$dir"
+         cp -r "$sqllogictest_checkout"/test/"$fd" "$TMP_TESTING_DIR"/"$fd"
+    done
+}
+
+function checkout_branch_if_exists() {
+  local cmd="$1"
+  local branch_name="$2"
+  local hash="$3"
+
+  if [ `"$cmd" branch --list "$branch_name" | wc -l` -eq 1 ]; then
+    "$cmd" checkout "$branch_name"
+  else
+    if [ -z "$hash" ]; then
+      "$cmd" checkout -b "$branch_name";
+    else
+      "$cmd" checkout -b "$branch_name" "$hash";
+    fi
+  fi
+}
```

Comment:
this is the function im looking for... In the `git` case I _do_ want to checkout at a particular hash, i _do_ _not_ want to checkout at a hash in the `dolt` case